spylon_kernel/scala_magic.py

```diff
@@ -190,21 +230,38 @@ def trim(prefix, completions):
 
         prefix = info['code'][info['start']:info['help_pos']]
 
-        offset = trim(prefix, c)
+        offset = trim(prefix, completions)
 
-        a = [prefix + h[offset:] for h in c]
-        self.kernel.log.debug("info %s\n    completions %s\n     final %s", info, c, a)
-        return a
+        final_completions = [prefix + h[offset:] for h in completions]
+        self.kernel.log.debug('''info %s
+            completions %s
+            final %s''', info, completions, final_completions)
+        return final_completions
 
     def get_help_on(self, info, level=0, none_on_fail=False):
+        """Gets help text for the `info['help_obj']` identifier.
+
+        Part of the MetaKernel API.
+
+        Parameters
+        ----------
+        info : dict
+            Information returned by `metakernel.parser.Parser.parse_code`
+            including `help_obj`, etc.
+        level : int
+            Level of help to request, 0 for basic, 1 for more, etc.
+        none_on_fail : bool
+            Ignored
+
+        Returns
+        -------
+        str
+            Help text
+        """
         intp = self._get_scala_interpreter()
         self.kernel.log.debug(info['help_obj'])
         # Calling this twice produces different output
```

Comment:
Basically it's pretty heavily assumed that this is used in the scala repl. So it is the same call for two tabs.  It is kinda mad